### DISCUSSION TITLE
Class "...\Pages\ListItem" does not exist after running v3 migration script

### DISCUSSION THREAD

**Discussion by pekka** [filamentphp/filament#7472] [Help] [answered]
This is mainly to serve as a piece of information for others who run into the same problem. I don't know if this is worth investigating further as a bug (but happy to start digging if necessary). 

I'm upgrading an existing project from v2 to v3 and can't run the `php artisan filament:install` step because it keeps crashing with messages like:

>   Class "App\Filament\Resources\CustomerResource\Pages\ListItem" does not exist

This is apparently because the upgrade script didn't apply the new naming convention (from `CreateItem.php` to `Create<ResourceType>.php` etc.) to all the resources that I have defined:

<img width="246" alt="image" src="https://github.com/filamentphp/filament/assets/262464/5cf99897-79b0-4512-b608-367c579944fe">

I have an inkling this could be because one resource type was literally named `Item` (marked orange in the screenshot) and maybe that confused the upgrade script. I don't understand Rector enough to tell.

Anyway, the fix appears to be to do the remaining renames manually:

<img width="250" alt="image" src="https://github.com/filamentphp/filament/assets/262464/1a20633e-940c-4f15-8b1f-5f5ad69bae36">

remember to also adjust the class names:

![image](https://github.com/filamentphp/filament/assets/262464/753d022e-5634-4c5e-be30-ac604888c826)

and the references in getPages() in the resource's main file:

![image](https://github.com/filamentphp/filament/assets/262464/bf1cc388-eb85-4570-b717-6d772e0dccc6)

**Comment by danharrin** [comment, marked as answer]
Um, I've not seen that `*Item` naming convention before, its not v2? And even so, the pages dont need to be named a certain way? As long as they are referenced correctly in `getPages()`. Something weird is going on here

**Reply by pekka** [reply]
Ooohhh, this must be the result of me copy+pasting resources when creating the project back in the day 😬 I thought this was something built in. Please disregard.
